core/src/cubos/core/io/keyboard.cpp

```diff
@@ -0,0 +1,119 @@
+#include <cubos/core/data/deserializer.hpp>
+#include <cubos/core/data/serializer.hpp>
+#include <cubos/core/io/keyboard.hpp>
+
+using cubos::core::data::Deserializer;
+using cubos::core::data::Serializer;
+using cubos::core::io::Key;
+using cubos::core::io::Modifiers;
+using namespace cubos::core;
+
+std::string io::modifiersToString(Modifiers modifiers)
+{
+    std::string result;
+
+    if ((modifiers & Modifiers::Control) != Modifiers::None)
+        result += "C-";
+    if ((modifiers & Modifiers::Shift) != Modifiers::None)
+        result += "S-";
+    if ((modifiers & Modifiers::Alt) != Modifiers::None)
+        result += "M-";
+    if ((modifiers & Modifiers::System) != Modifiers::None)
+        result += "D-";
+
+    return result;
+}
+
+template <>
+void cubos::core::data::serialize<Modifiers>(Serializer& ser, const Modifiers& obj, const char* name)
+{
+    ser.write(io::modifiersToString(obj), name);
+}
+
+Modifiers io::stringToModifiers(const std::string& str)
+{
+    Modifiers result = Modifiers::None;
+
+    if (str.find("C-") != std::string::npos)
+        result |= Modifiers::Control;
+    if (str.find("S-") != std::string::npos)
+        result |= Modifiers::Shift;
+    if (str.find("M-") != std::string::npos)
+        result |= Modifiers::Alt;
+    if (str.find("D-") != std::string::npos)
+        result |= Modifiers::System;
+
+    return result;
+}
+
+template <>
+void cubos::core::data::deserialize<Modifiers>(Deserializer& des, Modifiers& obj)
+{
+    std::string str;
+    des.read(str);
+    obj = io::stringToModifiers(str);
+}
+
+std::string io::keyToString(Key key)
+{
+    char key_char = static_cast<char>(key);
+
+    if (key_char >= 0 && key_char <= 25)
+    {
+        return std::string(1, key_char + 'a');
+    }
```

Comment:
I think its best not to do this, just write a `switch` for every key - way less error prone and if the enum changes for some reason it won't break .